func-mini-lib.js

```diff
@@ -0,0 +1,90 @@
+let arrLib = {};
+
+arrLib = (function (context) {
+
+    context.isArray = function (obj) {
+        return Array.isArray(obj);
+    };
+
+    context.isBoolean = function (obj) {
+        return typeof obj === 'boolean' ||
+            obj instanceof Boolean;
+    };
+
+    context.isDate = function (obj) {
+        return obj instanceof Date;
+    };
+
+    context.isNumber = function (obj) {
+        return !isNaN(parseFloat(obj)) && isFinite(obj);
+    };
+
+    context.isString = function (obj) {
+        return typeof obj === "string" ||
+            obj instanceof String;
+    };
+
+    context.isFunction = function (obj) {
+        return typeof obj === "function";
+    };
+
+    context.isUndefined = function (obj) {
+        return typeof obj === "undefined";
+    };
+
+    context.isNull = function (obj) {
+        return obj === null;
+    };
+
+    let value;
```

Comment:
You mixed two different parts into a one context. Chaining, array lib and types checks should be in a separate modules.